### DISCUSSION TITLE
Queue not working

### DISCUSSION THREAD

**Discussion by w99910** [jenssegers/laravel-mongodb#2222] [Q&A] [answered]
- Laravel-mongodb Version: 3.8
- PHP Version: 7.4.3
- Laravel Version:8.12

### Queue.php
`  'database' => [`
           ` 'driver' => 'mongodb',`
           ` 'table' => 'jobs',`
          `  'connection' => 'mongodb',`
            `'queue' => 'default',`
          `  'expire' => 60,`
           ` 'retry_after' => 90`
       ` ],`
### database.php
`   'mongodb' => [`
           ` 'driver' => 'mongodb',`
            `'host' => env('DB_HOST', '127.0.0.1'),`
            `'port' => env('DB_PORT', 27017),`
            `'database' => env('DB_DATABASE'),`
            `'username' => env('DB_USERNAME'),`
           ` 'password' => env('DB_PASSWORD'),`
          `  'options' => [`
          `  ],`
       ` ],`
I have added `Jenssegers\Mongodb\MongodbQueueServiceProvider::class,` inside `config/app.php`.

### Controller
` \App\Jobs\SampleJob::dispatch()->onConnection('database');`

### App\Jobs\SampleJob.php
I just info('test') inside handle function.

Job is created but never attempted.
I have already run `php artisan queue:listen`. But it didn't work.

![Screenshot from 2021-03-13 14-28-48](https://user-images.githubusercontent.com/65523475/111023644-9a675480-8408-11eb-8a7a-48da05e4ba45.png)

**Comment by GromNaN** [comment]
https://jira.mongodb.org/browse/PHPORM-87

**Reply by firrae** [reply]
Is there any progress on this? This is a breaking issue for me and a reason I'm moving back to Postgres at this point as job queues just seem unusable, let alone the new cache issues due to upsert.

**Comment by bisht2050** [comment, marked as answer]
Queues support was added in v4.3 
https://www.mongodb.com/docs/drivers/php/laravel-mongodb/current/queues/
